leo_fw/scripts/calibrate_imu

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+import argparse
+from calendar import day_abbr
+
+import numpy as np
+
+import rclpy
+from rclpy.node import Node
+from rclpy.qos import QoSProfile, QoSDurabilityPolicy, QoSReliabilityPolicy
+from leo_msgs.srv import SetImuCalibration
+from leo_msgs.msg import Imu
+
+
+class Calibrator(Node):
+    def __init__(self, time=2.0):
+        super().__init__("calibrate_imu")
+
+        self.service_client = self.create_client(
+            SetImuCalibration, "set_imu_calibration"
+        )
+
+        while not self.service_client.wait_for_service(timeout_sec=1.0):
+            self.get_logger().info("service not available, waiting again...")
+
+        self.data = []
+        self.end_time = self.get_clock().now() + rclpy.duration.Duration(seconds=time)
+
+        qos = QoSProfile(
+            depth=1,
+            durability=QoSDurabilityPolicy.VOLATILE,
+            reliability=QoSReliabilityPolicy.BEST_EFFORT,
+        )
+        self.imu_sub = self.create_subscription(
+            Imu, "firmware/imu", self.imu_sub_callback, qos
+        )
+
+        self.running = True
```

Comment:
Move this before creating the subscription to avoid a possible race condition.

Reply:
Done